app/models/mixins/process_tasks_mixin.rb

```diff
@@ -119,9 +119,13 @@ def invoke_api_tasks(api_client, remote_options)
 
       if resource_ids.present?
         resource_ids.each do |id|
-          obj = collection.find(id)
-          _log.info("Invoking task #{action} on collection #{collection_name}, object #{obj.id}, with args #{post_args}")
-          obj.send(action, post_args)
+          begin
+            obj = collection.find(id)
+            _log.info("Invoking task #{action} on collection #{collection_name}, object #{obj.id}, with args #{post_args}")
+            obj.send(action, post_args)
+          rescue RuntimeError => err
+            raise unless err.message =~ /couldn't find resource with 'id' \["\d+"\]/i
```

Comment:
This regex feels too specific and the message could change a little and suddenly break this.  Does it raise a useful exception class we can rescue instead?  If not, is `/couldn't find resource with 'id'/i` enough?

Reply:
It does not raise a useful exception class.  That's the entire error message.

Reply:
I feel like we want to raise a UnretryableException or something like that (better name) and have `invoke_tasks_remote` catch it, log it and move on.  I think swallowing it here hides the fact that we're explicitly skipping re-queueing.

Reply:
Maybe we can rescue just aroung the `collection.find(id)` call on line 123?

Then we can avoid the conditional entirely I think.

Reply:
Maybe?

```ruby
obj = collection.find_by(:id => id)
if obj
  _log.info("Invoking task #{action} on collection #{collection_name}, object #{obj.id}, with args #{post_args}")
  obj.send(action, post_args)
end
```

Reply:
@kbrock does the API client return `nil` for `find_by` rather than an exception?

Reply:
@carbonin aah - not sure. I was assuming it followed AR conventions - which is probably wrong
thanks